docs/promtail.md

```diff
@@ -5,6 +5,7 @@
   * [Deployment Methods](./promtail-setup.md)
   * [Promtail API](./promtail-api.md)
   * [Config and Usage Examples](./promtail-examples.md)
+  * [Failure modes](./promtail-failure-modes.md)
```

Comment:
Can you put this file in a `docs/promtail` folder, I think the goal with another PR is to move all the promtail specific docs into a `promtail` folder